### DISCUSSION TITLE
Infinite re-render when putting non-serializable objects in entities

### DISCUSSION THREAD

**Discussion by bgoosmanviz** [arnelenero/simpler-state#26] [General]
This code works as long as I pick a subset of values out of the object. The `extension` object is non-serializable, with functions in it, in addition to strings and numbers. 

```
// extension.ts
export const extension = entity<any | undefined>(getExtension(), [logger("extension")]);

async function getExtension() {
  const extension = await initExtension({ scope: ExtensionScope.ReadWrite, validateOrigin: () => true });
  // This works 👇
  return _.pick(extension, ["file", "params", "scope", "user", "container"]);
  // This causes an infinite loop 👇
  return extension;
}

// App.tsx
export function App() {
  const extensionJson = extension.use();
  ...
  console.log("☠️ infinite loop");
}
```

**Comment by arnelenero** [comment]
Hi. Thanks for posting.

Would you care to share the implementation of your `logger` plug-in?

**Comment by bgoosmanviz** [comment]
import { Plugin } from "simpler-state";

export const logger = (name: string): Plugin => {
  return {
    set:
      (origSet, entity) =>
      (...args) => {
        const prev = entity.get();
        origSet(...args);

        console.log(`${name}:`, prev, "-->", entity.get());
      },

    init: (origInit, entity) => () => {
      origInit();

      console.log(`[${name}] Initial value:`, entity.get());
    },
  };
};
